presto-main/src/main/java/com/facebook/presto/server/HttpRequestSessionContext.java

```diff
@@ -268,6 +272,27 @@ private static void assertRequest(boolean expression, String format, Object... a
         }
     }
 
+    private static Map<String, String> parseSessionFunctionHeader(HttpServletRequest req)
```

Comment:
Once maps are deserialized from the http request. I don't think we have a reason to keep them as Map<String, String>. We should just make it return `Map<BasicSignature, SqlInvokedFunction>` here.

Reply:
Agreed, done